Create job to bump testgrid config

chases2 opened the pull request:
/hold
for necessary credentials

transfigure.sh automates the process of creating a testgrid yaml file and pushing it to the appropriate k8s/test-infra folder.

It can be put into a Prow job, as demonstrated, to have a bot maintain a PR that automatically updates whenever the source files are changed.

/cc @clarketm @howardjohn @fejta

cjwagner commented on testgrid/transfigure.sh:
```diff
@@ -0,0 +1,27 @@
+export CONFIG_PATH="${root}/prow/config.yaml"
+export JOB_CONFIG_PATH="${root}/prow/cluster/jobs"
+export TESTGRID_CONFIG="${root}/testgrid/default.yaml"
+export ORG_REPO="istio"
```
nit: I would expect this to be `istio/test-infra` based on the variable name.

cjwagner commented on prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml:
```diff
@@ -92,6 +92,30 @@ postsubmits:
+        command:
+        - ./testgrid/transfigure.sh
+        args:
+        - /etc/github-token
```
I suspect this should be `/etc/github-token/oauth`

chases2 replied:
Fixed

fejta commented on testgrid/transfigure.sh:
```diff
@@ -0,0 +1,27 @@
+export ORG_REPO="istio"
+
+# Get the common instance of transfigure.sh and run it
+bash <(curl -s https://raw.githubusercontent.com/kubernetes/test-infra/master/testgrid/transfigure.sh) "$@"
```
I think a better pattern for this will be to create a gcr.io/k8s-prow/transfigure or whatever image that runs this script

https://github.com/kubernetes/test-infra/blob/0492f1d6ebc23b10ebe8a756e81fcbacf85ac92a/prow/cmd/autobump/BUILD.bazel

https://github.com/istio/test-infra/blob/4cba48e9bbe4093f3770813f3c841866fc1c51e8/prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml#L128-L167

This is how @cjwagner did all the autobumper jobs. Maybe ask him to walk you through the design if reading through the above isn't obvious?

fejta reviewed (commented):
Instead of copying the transfigure script everywhere, can we create a transfigure image that has the script inside it? 

This is how the bump jobs work, for example.

chases2 committed:
Add "post-test-infra-upload-testgrid-config" job

Job maintains a PR in k8s/test-infra that contains the TestGrid config
of this repository

fejta reviewed (approved):
/hold
